auslib/db.py

```diff
@@ -2395,6 +2398,23 @@ def hasRole(self, username, role, transaction=None):
         roles_list = [r['role'] for r in self.getUserRoles(username, transaction)]
         return role in roles_list
 
+    def isKnownUser(self, username):
+        if not username:
+            return False
+        cache_column = 'username'
+        cache_name = 'users'
+        cache_key = 'usernames'
+
+        def user_getter():
+            permissions = self.select(columns=[cache_column], distinct=True)
+            return [permission[cache_column] for permission in permissions]
+
+        if cache_name not in cache.caches:
+            cache.make_cache(cache_name, 1, 300)
```

Comment:
It would be better to do this when the app starts up (like the other caches in uwsgi/admin.wsgi) instead of on the fly -- it makes it more clear what caches we have in the app.